src/library/dao/trackdao.cpp

```diff
@@ -1510,7 +1472,6 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
     // of the properties has finished.
 
     // For every column run its populator to fill the track in with the data.
-    bool shouldDirty = false;
```

Comment:
I don't get the rationale for removing this.

Reply:
This is no longer used . The rational is described here: https://github.com/mixxxdj/mixxx/commit/1fc3e7477fc66cb273afd13348c0cb77a34e0280

Reply:
>  No longer set tracks dirty after reading form a legacy database.
> 
> This was only used in setTrackKey() to replace a missing source flag with USER.
Even if the database is from an earlier version of Mixxx the data is consistent and the track can imported every time again.

But won't that mean the in the non-legacy database that the source flag will be set to an undefined value or missing?

Reply:
The adopting of the USER state in RAM in this case remains. 
In the current released version, the missing keySource is set to USER in the Track Object and marked dirty which forces a write to track metadata and database. This writing is unnecessary, because Mixxx will adopt the USER state anyway again after restart.

Reply:
Ah I see. So the update is just postponed to next "flush" to the database?

Reply:
Yes, when anything else dirties the track.